pkg/mount-manager/mount-manager.go

```diff
@@ -19,88 +17,106 @@ package mountmanager
 import (
 	"fmt"
 	"os"
+	"path"
 	"path/filepath"
 	"strings"
 
 	"github.com/golang/glog"
-	compute "google.golang.org/api/compute/v1"
 	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/kubernetes/pkg/util/mount"
 	"k8s.io/utils/exec"
 )
 
-func matchDiskMode(disk *compute.AttachedDisk, readonly bool) bool {
-	if (disk.Mode == "READ_ONLY" && !readonly) || (disk.Mode == "READ_WRITE" && readonly) {
-		return false
-	}
-	return true
+const (
+	diskByHostIdPath     = "/host/dev/disk/by-id/"
+	diskByIdPath         = "/dev/disk/by-id/"
+	diskGooglePrefix     = "google-"
+	diskScsiGooglePrefix = "scsi-0Google_PersistentDisk_"
+	diskPartitionSuffix  = "-part"
+	diskSDPath           = "/host/dev/sd"
+	diskSDPattern        = "/host/dev/sd*"
+	// How many times to retry for a consistent read of /proc/mounts.
+	maxListTries = 3
+	// Number of fields per line in /proc/mounts as per the fstab man page.
+	expectedNumFieldsPerLine = 6
+	// Location of the mount file to use
+	procMountsPath = "/proc/mounts"
+	// Location of the mountinfo file
+	procMountInfoPath = "/proc/self/mountinfo"
+	// 'fsck' found errors and corrected them
+	fsckErrorsCorrected = 1
+	// 'fsck' found errors but exited without correcting them
+	fsckErrorsUncorrected = 4
+	defaultMountCommand   = "mount"
+)
+
+type DeviceUtils interface {
+	GetDiskByIdPaths(pdName string, partition string) []string
+	// TODO: Info
+	VerifyDevicePath(devicePaths []string) (string, error)
 }
 
-// GetDiskFormat uses 'lsblk' to see if the given disk is unformated
-func getDiskFormat(disk string) (string, error) {
-	args := []string{"-n", "-o", "FSTYPE", disk}
-	glog.V(4).Infof("Attempting to determine if disk %q is formatted using lsblk with args: (%v)", disk, args)
-	dataOut, err := execRun("lsblk", args...)
-	output := string(dataOut)
-	glog.V(4).Infof("Output: %q", output)
+// TODO: Info
+type deviceUtils struct {
+}
 
-	if err != nil {
-		glog.Errorf("Could not determine if disk %q is formatted (%v)", disk, err)
-		return "", err
+func NewSafeMounter() *mount.SafeFormatAndMount {
+	realMounter := mount.New("")
+	realExec := mount.NewOsExec()
+	return &mount.SafeFormatAndMount{
+		Interface: realMounter,
+		Exec:      realExec,
 	}
 
-	// Split lsblk output into lines. Unformatted devices should contain only
-	// "\n". Beware of "\n\n", that's a device with one empty partition.
-	output = strings.TrimSuffix(output, "\n") // Avoid last empty line
-	lines := strings.Split(output, "\n")
-	if lines[0] != "" {
-		// The device is formatted
-		return lines[0], nil
-	}
+}
 
-	if len(lines) == 1 {
-		// The device is unformatted and has no dependent devices
-		return "", nil
-	}
+var _ DeviceUtils = &deviceUtils{}
```

Comment:
Can you put the DeviceUtils into a separate file?

Reply:
done